src/pages/MainPage.vue

```diff
@@ -179,22 +143,13 @@
         >
           <road
             :selected-subjects="roads[roadId].contents.selectedSubjects"
-            :subjects="subjectsInfo"
             :road-i-d="roadId"
             :current-semester="currentSemester"
             :adding-from-card="addingFromCard && activeRoad===roadId"
             :item-adding="itemAdding"
-            :subjects-index="subjectsIndexDict"
-            :generic-courses="genericCourses"
-            :generic-index="genericIndexDict"
             :drag-semester-num="(activeRoad===roadId) ? dragSemesterNum : -1"
             @add-at-placeholder="addAtPlaceholder"
-            @add-class="addClass"
-            @move-class="moveClass($event.classIndex,$event.semester)"
-            @remove-class="removeClass"
-            @click-class="pushClassStack($event.id)"
             @change-year="$refs.authcomponent.changeSemester($event)"
-            @override-warnings="overrideWarnings($event.override,$event.classInfo)"
             @drag-start-class="dragStartClass"
```

Comment:
This function also seems to have many emits, so would benefit from being moved into the store. These are optimizations that we may want to deal with after this PR, though.

Reply:
Hmm, now that you mention it, `dragStartClass` is emitted across the board a fair bit. I can probably do the work as part of this PR.